src/util/gobConfig.ml

```diff
@@ -75,7 +75,17 @@ sig
   (** Merge configurations from a JSON object with current. *)
   val merge : Yojson.Safe.t -> unit
 
-  val json_conf: Yojson.Safe.t ref
+  (* TODO: can bypass immutability by directly setting value *)
+  val json_conf : Yojson.Safe.t ref
```

Comment:
Currently `merge` from this file also bypasses the check. It might be best to not expose `json_conf` at all but just have `set_json_conf` or something. That could just be directly used inside `GobConfig` as well, so the same check applies automatically also to single option set commands.